components/AlertsDashboard.vue

```diff
@@ -119,6 +123,43 @@ onMounted(() => {
       calculateHectares.value = true;
     }
     showSlider.value = true;
+
+    // Check for alertId in URL and select the corresponding alert
+    const alertId = route.query.alertId as string;
+    if (alertId) {
+      const allFeatures = [
+        ...props.alertsData.mostRecentAlerts.features,
+        ...props.alertsData.previousAlerts.features,
+      ];
+      const feature = allFeatures.find(
+        (f) => f.properties?.alertID === alertId,
+      );
+      if (feature) {
+        // Find the appropriate layer ID for this feature
+        const layerId = feature.properties?.isRecent
+          ? "most-recent-alerts-polygon"
+          : "previous-alerts-polygon";
+        selectFeature(feature, layerId);
+
+        // Zoom to the feature
+        if (feature.geometry.type === "Point") {
+          const [lng, lat] = feature.geometry.coordinates;
+          map.value.flyTo({ center: [lng, lat], zoom: 13 });
+        } else if (
+          feature.geometry.type === "Polygon" ||
+          feature.geometry.type === "MultiPolygon"
+        ) {
+          const bounds = bbox(feature);
+          map.value.fitBounds(bounds, { padding: 50 });
+        } else if (feature.geometry.type === "LineString") {
+          const line = lineString(feature.geometry.coordinates);
+          const lineLength = length(line, { units: "kilometers" });
+          const midpoint = along(line, lineLength / 2, { units: "kilometers" });
+          const [lng, lat] = midpoint.geometry.coordinates;
```

Comment:
I could have sworn we had a `calculateLinestringCentroid()` function (or something similar) somewhere!

But since we don't seem to, and this code is also repeated in `addPulsingMarker()`, we can make it one.